tests/console/cups.pm

```diff
@@ -0,0 +1,73 @@
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test basic capabilities of cups
+# Maintainer: Jan Baier <jbaier@suse.cz>
+
+use 5.018;
+use warnings;
+use base 'consoletest';
+use testapi;
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    script_run 'echo FileDevice Yes >> /etc/cups/cups-files.conf';
+    validate_script_output 'cupsd -t', sub { m/is OK/ };
+    assert_script_run 'systemctl enable cups.service';
+    assert_script_run 'systemctl start cups.service';
+    validate_script_output 'systemctl status cups.service', sub { m/Active:\s*active/ };
+
+    # Add printers
+    record_info "lpadmin", "Try to add printers and enable them";
+    validate_script_output 'lpstat -p -d -o 2>&1 || test $? -eq 1', sub { m/lpstat: No destinations added/ };
+    assert_script_run 'lpadmin -p printer_tmp -v file:/tmp/test_cups -m raw -E';
+    assert_script_run 'lpadmin -p printer_null -v file:/dev/null -m raw -E';
+    assert_script_run 'cupsenable printer_tmp printer_null';
+    assert_script_run 'lpoptions -d printer_tmp';
+    validate_script_output 'lpstat -p -d -o', sub { m/printer_tmp is idle/ };
+
+    assert_script_run 'wget ' . data_url('console/sample.ps');
+
+    # Submit print job to the queue, list them and cancel them
+    record_info "lp, lpstat, cancel", "Submitting and canceling jobs";
+    foreach my $printer (qw/printer_tmp printer_null/) {
+        assert_script_run "cupsdisable $printer";
+        assert_script_run "lp -d $printer -o cpi=12 -o lpi=8 sample.ps";
+        validate_script_output 'lpstat -o',          sub { m/$printer-\d+/ };
+        validate_script_output 'ls /var/spool/cups', sub { m/d\d+/ };
+        assert_script_run "cancel -a $printer";
+    }
+
+    assert_script_run 'systemctl restart cups.service';
+    validate_script_output 'systemctl status cups.service', sub { m/Active:\s*active/ };
+
+    # Do the printing
+    record_info "lp, lpq", "Printing jobs";
+    foreach my $printer (qw/printer_tmp printer_null/) {
+        assert_script_run "cupsenable $printer";
+        assert_script_run "lp -d $printer /usr/share/cups/data/default-testpage.pdf";
+        validate_script_output "lpq $printer", sub { m/is ready/ };
+    }
+
+    # Check logs
+    record_info "access_log", "Access log should containt successful job submits";
+    assert_script_run 'grep "Send-Document successful-ok" /var/log/cups/access_log';
+
+    # Remove printers
+    record_info "lpadmin -x", "Removing printers";
+    assert_script_run "lpadmin -x $_" foreach (qw/printer_tmp printer_null/);
+    validate_script_output 'lpstat -p -d -o 2>&1 || test $? -eq 1', sub { m/No destinations added/ };
+    assert_script_run 'systemctl stop cups.service';
+    assert_script_run 'systemctl disable cups.service';
```

Comment:
follow up on comment below :
```suggestion
    my $out = script_run 'systemctl disable cups.service';
    die "Fail to disable cups" unless $out eq 0 or $out eq 3;
```
note : this code might have "grammar" issues but you can get the idea

Reply:
I do not have problem with `assert_script_run` in this case, the problem is `validate_script_output` (which uses the assert itself).